sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVInferSchema.scala

```diff
@@ -233,7 +238,39 @@ class CSVInferSchema(val options: CSVOptions) extends Serializable {
    * is compatible with both input data types.
    */
   private def compatibleType(t1: DataType, t2: DataType): Option[DataType] = {
-    TypeCoercion.findTightestCommonType(t1, t2).orElse(findCompatibleTypeForCSV(t1, t2))
+    (t1, t2) match {
+      case (DateType, TimestampType) | (DateType, TimestampNTZType) |
+           (TimestampNTZType, DateType) | (TimestampType, DateType) =>
+        // For a column containing a mixture of dates and timestamps
+        // infer it as timestamp type if its dates can be inferred as timestamp type
+        // otherwise infer it as StringType
```

Comment:
I don't quite understand the rationale here. why can't we directly return `Some(StringType)`?

Reply:
We want to have consistent behavior when timestamp format is not given.

When `prefersDate=false`, a column with mixed date and timestamp could be inferred as timestamp if possible.
Thus, we added the additional handling here to have similar behavior as above when `prefersDate=true`.

Does this make sense?

Reply:
nvm, I got it